advent-of-code/rucksack-problem/src/rucksack_problem/core.clj

```diff
@@ -0,0 +1,136 @@
+;; Advent of Code Day 03
+
+;; Part 1
+;;      Process a single rucksack
+;;   1. Divide rucksack string into 2 halves
+;;   2. Check for the common item (only one exists) in both halves
+;;   3. Find the priority of the common item using the below relationships:
+;;      Priority of a - z: 1 - 26
+;;      Priority of A - Z: 27 - 52)
+;;
+;;      And then expand to multiple rucksacks to solve Part 1
+
+;; Lifting some input restrictions as another exercise
+;; Lift restriction of a single common item. Code generic solution where priorities are counted
+;; for each pair of common items between the 2 compartments.
+;; 1. create hashmap for compartment 1 and 2 with key: item, value: frequency of items
+;; 2. find pairs of common keys between both hashmaps
+;; 3. minimum value of each key pair is the priority multiple
+;; 4. sum of (priority * multiple) for each key pair
+
+;; Part 2
+;; For 1 group of 3 rucksacks
+;; 1. find common item in each group of 3 rucksacks.
+;; 2. find priority of common item
+;; Repeat the above for each group of 3 rucksacks, and sum the priorities of common items.
+
+(ns rucksack-problem.core
```

Comment:
In general, try to avoid calling your namespace `core`. It doesn't say anything about what this module does.

In this case, it's not a big deal, since we are having only a single module, but this is something to keep in mind when you move to multi-namespace projects.

Reply:
Good point, I've been going with the default `core`. Would `solution` be a good name for this namespace?